lib/router/router.js

```diff
@@ -15,6 +19,10 @@ export default class Router extends EventEmitter {
 
     // set up the component cache (by route keys)
     this.components = { [this.route]: { Component, err } }
+    // contain a map of response of prefetched routes
+    this.prefetchedRoutes = {}
+    this.prefetchingLockManager = new LockManager(2)
+    this.prefetchingRoutes = {}
```

Comment:
@nkzawa we don't. But user could ask to do that. That's why we need to keep these states.

For an example, let's say user asking us to fetch following pages:

~~~js
<Link prefetch href="/abc?aa=10" as="/abc/10" />
<Link prefetch href="/abc?aa=20" as="/abc/20" />
...
~~~

But we only need to fetch `/abc` once. That's why.